### PR TITLE
support batch_size>1 for some operators

### PR DESCRIPTION
**Pull request by Cathy0908**

### CODE REVIEW

File: data_juicer/ops/base_op.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def __init__(self, *args, **kwargs):
         self.image_key = kwargs.get('image_key', 'images')
         self.audio_key = kwargs.get('audio_key', 'audios')
         self.video_key = kwargs.get('video_key', 'videos')
+        self.batch_size = kwargs.get('batch_size', 1)
```
Comment (drcege) [inline data_juicer/ops/base_op.py:136]: Should we change the default value to 1000 to align with the default values of HF `dataset.map/filter`?

**Review by drcege — COMMENTED** [review]
There are too many ops involved; everyone's effort is needed for the review!
@zhijianma @HYLcool @pan-x-c

**Comment by drcege** [comment]
I greatly appreciate the contributions made in this PR. Please sync the latest changes from the `main` branch and carefully resolve any conflicts, particularly in `core/data.py`.

File: data_juicer/ops/base_op.py (hunk 1 above)
Comment (drcege) [reply data_juicer/ops/base_op.py:136]: Then 1000 might be better to align with users' expectations of existing tools.

File: data_juicer/ops/filter/alphanumeric_filter.py [hunk 2]
```diff
@@ -54,36 +56,49 @@ def __init__(self,
...
-        if self.min_ratio <= ratio <= self.max_ratio:
-            return True
+    def process(self, samples):
+        ratio_key = StatsKeys.alpha_token_ratio if self.tokenization \
+            else StatsKeys.alnum_ratio
+        if isinstance(samples[Fields.stats], list):
+            bool_results = []
+            for stat in samples[Fields.stats]:
+                if self.min_ratio <= stat[ratio_key] <= self.max_ratio:
+                    bool_results.append(True)
```
[the remark this answers is not in the document]
Comment (BeachWang) [reply data_juicer/ops/filter/alphanumeric_filter.py]: Have changed them by using `map`. Please check again @drcege

[the remark this answers is not in the document]
Comment (BeachWang) [reply data_juicer/ops/filter/alphanumeric_filter.py]: Must use `list(map())`, since `map` return a iterator in Python3.

**Review by yxdyc — APPROVED** [review]
TODO:
[] default `bs_size= 1`--> a larger suitable value
[] check redundant `_batched_op=True` in `process func`

**Pull request merged**